integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextbest/NextBestServiceBus.java

```diff
@@ -0,0 +1,145 @@
+                props.getNextbest().getServiceBus().getNamespace(),
+                "RootManageSharedAccessKey",
+                props.getNextbest().getServiceBus().getSasToken(),
+                ".servicebus.windows.net"
+        );
+        this.service = ServiceBusService.create(config);
+
+        this.jaxbContext = JAXBContext.newInstance(EduDocument.class, Payload.class);
+
+    }
+
+    @PostConstruct
+    private void init() throws ServiceException {
+
+        if (!props.getNextbest().getServiceBus().isEnable()) {
+            return;
+        }
+
+        // Create queue if it does not already exist
+        queuePath = String.format("%s%s%s", NEXTBEST_QUEUE_PREFIX,
+                props.getOrg().getNumber(),
+                props.getNextbest().getServiceBus().getMode());
+        ListQueuesResult queues = service.listQueues();
+        if (!queues.getItems().stream().anyMatch(i -> i.getPath().contains(queuePath))) {
+            log.info("Queue with id {} does not already exist, creating it..", queuePath);
+            QueueInfo qi = new QueueInfo(queuePath);
+            service.createQueue(qi);
+        }
+    }
+
+    public void putMessage(OutgoingConversationResource resource) throws NextBestException {
+
+        BrokeredMessage msg;
+        try {
+            MessageContext context = messageSender.createMessageContext(resource);
+            EduDocument eduDocument = sbdf.create(resource, context);
+
+            Marshaller marshaller = jaxbContext.createMarshaller();
+            ByteArrayOutputStream os = new ByteArrayOutputStream();
+            ObjectFactory of = new ObjectFactory();
+            JAXBElement<EduDocument> sbd = of.createStandardBusinessDocument(eduDocument);
+            marshaller.marshal(sbd, os);
+
+            msg = new BrokeredMessage(os.toByteArray());
+            os.close();
+
+            String queue = NEXTBEST_QUEUE_PREFIX + resource.getReceiverId();
+            if (ServiceIdentifier.DPE_INNSYN.fullname().equals(resource.getMessagetypeId())) {
+                queue = queue + "innsyn";
+            } else {
+                queue = queue + "data";
+            }
+            service.sendMessage(queue, msg);
+        } catch (ServiceException | MessageException | JAXBException | IOException e) {
+            log.error("Could not send conversation resource", e);
+            throw new NextBestException(e);
+        }
+
+    }
+
+    public List<EduDocument> getAllMessages() throws NextBestException {
+
+        ReceiveMessageOptions opts = ReceiveMessageOptions.DEFAULT;
+        opts.setReceiveMode(ReceiveMode.PEEK_LOCK);
+
+        ArrayList<EduDocument> messages = Lists.newArrayList();
+        try {
+            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+            for (;;) {
+                BrokeredMessage msg = service.receiveQueueMessage(queuePath, opts) .getValue();
+
+                if (msg == null || isNullOrEmpty(msg.getMessageId())) {
```

Comment:
Kanskje bytta ut for og denne if/break med en while((msg = service.receiveQueueMessage()) != null)